`make` doesn't actually work?

With `rebar` in `$PATH`:

```
$ make
==> covertool (get-deps)
==> covertool (update-deps)
make -C deps/rebar
make[1]: *** deps/rebar: No such file or directory. Stop.
Makefile:11: recipe for target 'deps' failed
make: *** [deps] Error 2
```

I had to `rebar compile && rebar escript`.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,13 @@
+sudo: false
+language: erlang
+otp_release:
+  - 19.3
+  - 20.3
+  - 21.3
+branches:
+  only:
+    - master
+    - /^[0-9]+\.[0-9]+\.[0-9]+/
+install: true
+script:
+  - make

--- a/Makefile
+++ b/Makefile
@@ -2,19 +2,12 @@
 
 .PHONY : all deps compile test clean
 all: deps compile test
-ifdef REBAR
+REBAR?=rebar
 deps:
-	$(info REBAR is $(REBAR))
-else
-REBAR=rebar
-deps:
-	@$(REBAR) get-deps update-deps
-	$(MAKE) -C deps/rebar
-endif
+	@$(REBAR) get-deps
 compile:
 	@$(REBAR) compile escriptize
 test:
 	-@$(REBAR) skip_deps=true eunit
 clean:
 	@$(REBAR) clean
-	@rm cover.xml

--- a/README.md
+++ b/README.md
@@ -1,5 +1,7 @@
 # Cover-to-Cobertura Conversion Tool 
 
+[![Build Status](https://travis-ci.org/nalundgaard/covertool.svg?branch=master)](https://travis-ci.org/nalundgaard/covertool)
+
 [![Hex.pm version](https://img.shields.io/hexpm/v/covertool.svg?style=flat)](https://hex.pm/packages/covertool)
 
 A simple tool to convert exported Erlang `cover` data sets into Cobertura XML
@@ -7,7 +9,7 @@ reports. The report could be then feed to the Jenkins Cobertura plug-in.
 
 ## Plugin rename notice
 
-This plugin is now known on [Hex](https://hex.pm) as [`covertool`](https://hex.pm/packages/covertool). Previously, is was called [`rebar_covertool`](https://hex.pm/packages/rebar_covertool). That package name is now **retired**, although it will remain compatible.
+This plugin is now known on [Hex](https://hex.pm) as [`covertool`](https://hex.pm/packages/covertool). Previously, it was called [`rebar_covertool`](https://hex.pm/packages/rebar_covertool). That package name is now **retired**, although it will remain compatible.
 
 As of `covertool` 1.3.0, the `covertool` Erlang module now contains callbacks for all supported plugin formats (rebar 2, rebar3, and mix). Going forward, referencing `covertool` as `rebar_covertool` or `mix_covertool` in plugins is **deprecated**. Use `covertool` instead (see [usage](#usage) below).
 

--- a/rebar.config
+++ b/rebar.config
@@ -1,6 +1,5 @@
 {erl_opts, [nowarn_untyped_record]}.
 {eunit_opts, [verbose, {report, {eunit_surefire, [{dir, "."}]}}]}.
-{deps,
-    [
-        {rebar, ".*", {git, "https://github.com/rebar/rebar.git", {tag, "2.6.0"}}}
-    ]}.
+{eunit_compile_opts, [export_all]}.
+{deps, []}.
+{post_hooks, [{clean, "rm -f cover.xml"}]}.

--- a/rebar.config.script
+++ b/rebar.config.script
@@ -1,9 +1,0 @@
-case os:getenv("REBAR") of
-    Missing when Missing =:= false; Missing =:= [] ->
-        %% we are not building covertool directly, so do *not* fetch rebar
-        {value, {deps, DepList}, RestConfig} = lists:keytake( deps, 1, CONFIG ),
-        NewDepList = lists:keydelete( rebar, 1, DepList ),
-        [{deps, NewDepList} | RestConfig];
-    _RebarCmd -> CONFIG
-end.
-

--- a/src/covertool.app.src
+++ b/src/covertool.app.src
@@ -1,7 +1,7 @@
 {application, covertool,
  [
   {description, "Build tool & plugin for generating Cobertura XML reports"},
-  {vsn, "2.0.0"},
+  {vsn, "2.0.1"},
   {registered, []},
   {applications, [kernel, stdlib]},
   {env, []},

--- a/test/rebar_covertool_tests.erl
+++ b/test/rebar_covertool_tests.erl
@@ -1,12 +1,6 @@
 -module(rebar_covertool_tests).
 
--compile(export_all).
-
 -include_lib("eunit/include/eunit.hrl").
-
-%% Assuming this test is run inside the rebar 'eunit'
-%% command, the current working directory will be '.eunit'
--define(REBAR_SCRIPT, "../deps/rebar/rebar").
 
 -define(TMP_DIR, "tmp_eunit/").
 
@@ -16,16 +10,19 @@
 
 eunit_test_() ->
     {"Ensure Cover runs with tests in a test dir and no defined suite",
-     setup, fun() -> setup_project(), rebar("-v compile eunit ct") end,
+     setup,
+     fun() ->
+        setup_project(),
+         rebar("-v compile eunit ct")
+     end,
      fun teardown/1,
-
      fun(RebarOut) ->
              [{"Plugin is found",
                ?_assert(string:str(RebarOut, "WARN:  Missing plugins:") =:= 0)},
-			  
-			  {"Plugin does not contain compilation errors",
-			   ?_assert(string:str(RebarOut, "contains compilation errors:") =:= 0)},
-			  
+
+              {"Plugin does not contain compilation errors",
+               ?_assert(string:str(RebarOut, "contains compilation errors:") =:= 0)},
+
               {"All cover reports are generated",
                assert_files_in("the temporary eunit directory",
                                expected_cover_generated_files())},
@@ -33,20 +30,20 @@ eunit_test_() ->
               {"Only production modules get coverage reports",
                assert_files_not_in("the temporary eunit directory",
                                    [".eunit/myapp_mymod_tests.COVER.html"])},
-			  
-			  {"myapp_mymod should be fully covered and package name must match!",
-			   assert_report(".eunit/eunit.coverage.xml", "otherapp")},
 
-			  {"myapp_mymod should be fully covered and package name must match!",
-			   assert_report("test/ct.coverage.xml", "otherapp")}
-			  ]
+              {"myapp_mymod should be fully covered and package name must match!",
+               assert_report(".eunit/eunit.coverage.xml", "otherapp")},
+
+              {"myapp_mymod should be fully covered and package name must match!",
+               assert_report("test/ct.coverage.xml", "otherapp")}
+              ]
      end}.
 
 expected_cover_generated_files() ->
     [".eunit/cover.coverdata",
-	 ".eunit/eunit.coverage.xml",
-	 "test/ct.coverdata",
-	 "test/ct.coverage.xml"].
+     ".eunit/eunit.coverage.xml",
+     "test/ct.coverdata",
+     "test/ct.coverage.xml"].
 
 %% ====================================================================
 %% Setup and Teardown
@@ -69,43 +66,42 @@ expected_cover_generated_files() ->
 
 -define(myapp_mymod_SUITE,
         ["-module(myapp_mymod_SUITE).\n",
-		 "-export([all/0]).\n",
-		 "-export([myfunc_test/1]).\n",
+         "-export([all/0]).\n",
+         "-export([myfunc_test/1]).\n",
          "-include_lib(\"common_test/include/ct.hrl\").\n",
          "all() -> [myfunc_test].\n",
          "myfunc_test(_Config) -> myapp_mymod:myfunc().\n"]).
 
 -define(rebar_config,
-		["{covertool_app_name, 'otherapp'}.\n",
-		 "{lib_dirs, [\"../../../\"]}.\n",
-		 "{plugins, [rebar_covertool]}.\n",
-		 "{cover_enabled, true}.\n",
-		 "{cover_export_enabled, true}.\n",
-		 "{covertool_eunit, {\".eunit/cover.coverdata\", \".eunit/eunit.coverage.xml\"}}.\n",
-		 "{covertool_ct, {\"test/ct.coverdata\", \"test/ct.coverage.xml\"}}.\n"]).
+        ["{covertool_app_name, 'otherapp'}.\n",
+         "{lib_dirs, [\"../../../\"]}.\n",
+         "{plugins, [rebar_covertool]}.\n",
+         "{cover_enabled, true}.\n",
+         "{cover_export_enabled, true}.\n",
+         "{covertool_eunit, {\".eunit/cover.coverdata\", \".eunit/eunit.coverage.xml\"}}.\n",
+         "{covertool_ct, {\"test/ct.coverdata\", \"test/ct.coverage.xml\"}}.\n"]).
 
 -define(cover_spec,
-		["{export, \"ct.coverdata\"}.\n",
-		"{incl_dirs, [\"../../test\", \"../../src\"]}.\n"]).
+        ["{export, \"ct.coverdata\"}.\n",
+         "{incl_dirs, [\"../../../test\", \"../../../src\"]}.\n"]).
 
 make_tmp_dir() ->
     ok = file:make_dir(?TMP_DIR).
 
 setup_environment() ->
     ok = make_tmp_dir(),
-    prepare_rebar_script(),
     ok = file:set_cwd(?TMP_DIR).
 
 setup_project() ->
-    setup_environment(),
+    ok = setup_environment(),
     rebar("create-app appid=myapp"),
     ok = file:make_dir("ebin"),
     ok = file:make_dir("test"),
     ok = file:write_file("test/myapp_mymod_tests.erl", ?myapp_mymod_tests),
     ok = file:write_file("src/myapp_mymod.erl", ?myapp_mymod),
-	ok = file:write_file("test/myapp_mymod_SUITE.erl", ?myapp_mymod_SUITE),
-	ok = file:write_file("test/cover.spec", ?cover_spec),
-	ok = file:write_file("rebar.config", ?rebar_config).
+    ok = file:write_file("test/myapp_mymod_SUITE.erl", ?myapp_mymod_SUITE),
+    ok = file:write_file("test/cover.spec", ?cover_spec),
+    ok = file:write_file("rebar.config", ?rebar_config).
 
 
 teardown(_) ->
@@ -113,31 +109,31 @@ teardown(_) ->
     ok = remove_tmp_dir().
 
 remove_tmp_dir() ->
-    remove_tmp_dir(arg_for_eunit).
-
-remove_tmp_dir(_) ->
     ok = rebar_file_utils:rm_rf(?TMP_DIR).
 
 %% ====================================================================
 %% Helper Functions
 %% ====================================================================
 
-prepare_rebar_script() ->
-    Rebar = ?TMP_DIR ++ "rebar",
-    {ok, _} = file:copy(?REBAR_SCRIPT, Rebar),
-    case os:type() of
-        {unix, _} ->
-            [] = os:cmd("chmod u+x " ++ Rebar);
-        {win32, _} ->
-            {ok, _} = file:copy(?REBAR_SCRIPT ++ ".bat",
-                                ?TMP_DIR ++ "rebar.bat")
+rebar_executable() ->
+    RebarName = case os:getenv("REBAR") of
+        Missing when Missing =:= false; Missing =:= [] -> "rebar";
+        CustomRebar -> CustomRebar
+    end,
+    find_rebar_executable(RebarName).
+
+find_rebar_executable(RebarName) ->
+    case filelib:is_file(RebarName) of
+        true -> RebarName;
+        false ->
+            case os:find_executable(RebarName) of
+                false -> erlang:error(missing_rebar, [RebarName]);
+                RebarExecutable -> RebarExecutable
+            end
     end.
 
-rebar() ->
-    rebar([]).
-
 rebar(Args) when is_list(Args) ->
-    Out = os:cmd(filename:nativename("./rebar") ++ " " ++ Args),
+    Out = os:cmd(rebar_executable() ++ " " ++ Args),
     Out.
 
 assert_files_in(Name, [File|T]) ->
@@ -152,9 +148,11 @@ assert_files_not_in(_, []) -> [].
 
 assert_report(File, AppName) ->
     fun() ->
-            {ok, F} = file:read_file(File),
-			Str = binary_to_list(F),
-			[?assert(string:str(Str, "filename=\"myapp_mymod.erl\" line-rate=\"1.000000\"") =/= 0),
-			 ?assert(string:str(Str, "package name=\"" ++ AppName ++ "\"") =/= 0)]
+        {ok, F} = file:read_file(File),
+        Str = binary_to_list(F),
+        Pattern = "<class name=\"myapp_mymod\" filename=\"[^\"]*myapp_mymod.erl\"[^<>]* line-rate=\"1.0",
+        Opts = [{capture, none}],
+        [?assertEqual(match, re:run(Str, Pattern, Opts)),
+         ?assert(string:str(Str, "package name=\"" ++ AppName ++ "\"") =/= 0)]
     end.
 

